cmd/dvotenode/dvotenode.go

```diff
@@ -543,6 +543,28 @@ func main() {
 			log.Fatal(err)
 		}
 
+		w3uris := []string{}
+		if len(globalCfg.W3Config.W3External) > 0 {
```

Comment:
I'd revert that logic... its a bit messy.

Remove the `if len(globalCfg.W3Config.W3External) > 0 {` and also the else and just add after the for

`if len(w3uris) == 0 { log.Fatal("no valid web3 endpoint found") }`